test/attribute/attribute_read_io_test.cpp

```diff
@@ -0,0 +1,150 @@
+//
+// (c) Copyright 2017 DESY,ESS
+//
+// This file is part of h5cpp.
+//
+// This library is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License as published
+// by the Free Software Foundation; either version 2.1 of the License, or
+// (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty ofMERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+// License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this library; if not, write to the
+// Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor
+// Boston, MA  02110-1301 USA
+// ===========================================================================
+//
+// Author: Jan Kotanski <jan.kotanski@desy.de>
+// Created on: Jun 19, 2018
+//
+#include <gtest/gtest.h>
+#include <h5cpp/hdf5.hpp>
+#include <string>
+
+using namespace hdf5;
+
+class AttributeReadIO: public testing::Test
+{};
+
```

Comment:
For these tests, maybe you could actually use the `EXPECT_EQ` or `ASSERT_EQ` to compare the written and read values, rather than print them out. See how the other tests do it. The tests should automatically validate the functionality and not require the user to read the output. You should not stream anything to `cout`.